src/encoding.rs

```diff
@@ -11,8 +50,9 @@ pub const BASE8: Encoding = new_encoding! {
     symbols: "01234567",
 };
 
+#[cfg(feature = "alloc")]
 /// Base10 (alphabet: 0123456789)
-pub const BASE10: &str = "0123456789";
+pub const BASE10: (&str, usize) = ("0123456789", 10);
```

Comment:
This change isn't needed, for the length you could just call `BASE10.len()`.

Reply:
`.len()` for slices is the total byte size but `.chars()` an iterator of inner chars, so `BASE10.chars().count()`?

Reply:
I don't mind either. All our current bases only take ASCII characters, which are all 1 byte. If we would add e.g. some base emoji encoding, this could  break I guess. So I think using `.len()` would be fine.

Though `.chars().count()` is more correct and I'm not really concerned about performance at the moment, so using it is also fine.

Choose your own preference :)